.github/workflows/build.yml

```diff
@@ -91,18 +91,91 @@ jobs:
     steps:
       - uses: actions/download-artifact@v4
         with:
-          pattern: artifact-bin-macos-*
+          pattern: unsigned-bin-macos-*
           merge-multiple: true
 
       - name: Link universal binary
         run: lipo -create -output sentry-cli-Darwin-universal sentry-cli-Darwin-x86_64 sentry-cli-Darwin-arm64
 
       - uses: actions/upload-artifact@v4
         with:
-          name: artifact-bin-macos-universal
+          name: unsigned-bin-macos-universal
           path: sentry-cli-Darwin-universal
           if-no-files-found: 'error'
 
+  sign-macos-binaries:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - arch: universal
+          - arch: x86_64
+          - arch: arm64
+
+    needs: [macos, macos_universal]
+    name: Sign & Notarize macOS Binary (${{ matrix.arch }})
+    runs-on: ubuntu-24.04
+
+    env:
+      APPLE_CERT_PATH: /tmp/certs.p12
+      APPLE_API_KEY_PATH: /tmp/apple_key.json
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install `rcodesign`
+        run: |
+          curl -L https://github.com/indygreg/apple-platform-rs/releases/download/apple-codesign%2F0.29.0/apple-codesign-0.29.0-x86_64-unknown-linux-musl.tar.gz | tar -xz --strip-components=1
+          mv rcodesign /usr/local/bin/rcodesign
```

Comment:
Recommend adding a step here that verifies the expected SHA256 checksum.

```
https://github.com/indygreg/apple-platform-rs/releases/download/apple-codesign%2F0.29.0/apple-codesign-0.29.0-aarch64-unknown-linux-musl.tar.gz.sha256
```

```
4af92c87ddf52f5f2d1258a3b4e56c7dcb8f1b2468df744976c5f139e031961f
```

Reply:
I think this is a good idea as I forgot releases can be mutable.

Reply:
Added to Spotlight here: https://github.com/getsentry/spotlight/pull/722/files

Reply:
Note that the SHA256 @mdtro shared is for arm64 so different in my PR.

Reply:
Oh woops. That's my mistake. 😅

Reply:
No worries, I made the same mistake even when I thought I was looking _very_ closely :D